app-modules/research/src/Actions/GenerateResearchQuestion.php

```diff
@@ -0,0 +1,111 @@
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Research\Actions;
+
+use AdvisingApp\Ai\Settings\AiIntegratedAssistantSettings;
+use AdvisingApp\Ai\Settings\AiResearchAssistantSettings;
+use AdvisingApp\Research\Models\ResearchRequest;
+use AdvisingApp\Research\Models\ResearchRequestQuestion;
+use App\Models\User;
+
+class GenerateResearchQuestion
+{
+    public function execute(ResearchRequest $researchRequest): string
+    {
+        return app(AiIntegratedAssistantSettings::class)
+            ->getDefaultModel()
+            ->getService()
+            ->complete(
+                prompt: $this->getPrompt($researchRequest),
+                content: $this->getContent($researchRequest),
+            );
+    }
+
+    protected function getContent(ResearchRequest $researchRequest): string
+    {
+        $questions = $researchRequest->questions
+            ->map(fn (ResearchRequestQuestion $question, int $index) => '**Question ' . ($index + 1) . ":** {$question->content}" . PHP_EOL . '**Answer ' . ($index + 1) . ":** {$question->response}")
+            ->implode(PHP_EOL . PHP_EOL);
+
+        if (filled($questions)) {
+            $questions = <<<EOD
+
+            
+                **Clarification Q & A:**
+
+                {$questions}
+                
+                ---
+                EOD;
+        }
+
+        return <<<EOD
+            **Research topic:**
+
+            {$researchRequest->topic}
+
+            ---{$questions}
+
+            **Instructions:**
+            
+            Before the research is conducted, you need to help improve the changes that it will be relevant and useful to me. Using my research topic, respond with one relevant question that will help to clarify it. You will be able to ask four questions in total, and you will know the answer to the previous question/s before you ask the next one. Do not respond with any greetings or salutations, and do not include any additional information or context. Just respond with your question:
```

Comment:
```suggestion
            Before the research is conducted, you need to help improve the chances that it will be relevant and useful to me. Using my research topic, respond with one relevant question that will help to clarify it. You will be able to ask four questions in total, and you will know the answer to the previous question/s before you ask the next one. Do not respond with any greetings or salutations, and do not include any additional information or context. Just respond with your question:
```

Reply:
done